src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/DateFormatTest.java

```diff
@@ -0,0 +1,29 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import org.junit.jupiter.api.Test;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class DateFormatTest {
+
+    @Test
+    void testFromCodeForValidCodeReturnsDateFormat() {
+        final ImmutableMap<String, DateFormat> codeMap = ImmutableMap.<String, DateFormat>builder()
+            .put("102", DateFormat.CCYYMMDD)
+            .put("602", DateFormat.CCYY)
+            .put("610", DateFormat.CCYYMM)
+            .build();
+
+        assertEquals(DateFormat.values().length, codeMap.size());
+        codeMap.forEach((code, dateFormat) -> assertEquals(dateFormat, DateFormat.fromCode(code)));
+    }
```

Comment:
Easier way to test all enum values if you're just verifying the `fromCode` method. Your way is nice if you also want the test to fail if we add more formats without revisiting the test.

```suggestion
    @ParameterizedTest
    @EnumSource(DateFormat.class)
    void testFromCodeForValidCodeReturnsDateFormat(final DateFormat format) {
        assertEquals(format, DateFormat.fromCode(format.getCode()));
    }
```

Reply:
Also applies to `Gender` and `PersonIdentificationType`

Reply:
This is nice, thanks!